python/paddle/fluid/layers/nn.py

```diff
@@ -676,10 +692,12 @@ def dynamic_gru(input,
         attr=helper.param_attr, shape=[size, 3 * size], dtype=dtype)
     bias = helper.create_parameter(
         attr=helper.bias_attr, shape=[1, 3 * size], dtype=dtype, is_bias=True)
+    batch_size = input.shape[0]
     inputs = {'Input': input, 'Weight': weight, 'Bias': bias}
     if h_0 != None:
         assert h_0.shape == (
-            size, size), 'The shape of h0 should be(%d, %d)' % (size, size)
+            batch_size, size
+        ), 'The shape of h0 should be(batch_size, %d)' % size
         inputs['h0'] = h_0
```

Comment:
It also should be `inputs['H0'] = h_0` like in LSTM. Sorry for leaving out this.

Reply:
Thanks! Done.